debian/patches/broadcom-firmware.patch

```diff
@@ -0,0 +1,152 @@
+ 			" [sleep|nosleep] [bdaddr]\n");
+ 	printf("\thciattach -l\n");
+@@ -1244,12 +1245,13 @@
+ 	struct pollfd p;
+ 	sigset_t sigs;
+ 	char dev[PATH_MAX];
++	char *firmware_dir = NULL;
+ 
+ 	detach = 1;
+ 	printpid = 0;
+ 	raw = 0;
+ 
+-	while ((opt=getopt(argc, argv, "bnpt:s:lr")) != EOF) {
++	while ((opt=getopt(argc, argv, "bnpt:s:lrf:")) != EOF) {
+ 		switch(opt) {
+ 		case 'b':
+ 			send_break = 1;
+@@ -1282,6 +1284,10 @@
+ 			raw = 1;
+ 			break;
+ 
++		case 'f':
++			firmware_dir = optarg;
++			break;
++
+ 		default:
+ 			usage();
+ 			exit(1);
+@@ -1358,6 +1364,8 @@
+ 		exit(1);
+ 	}
+ 
++	u->fwdir = firmware_dir;
++
+ 	/* If user specified a initial speed, use that instead of
+ 	   the hardware's default */
+ 	if (init_speed)
+Index: b/tools/hciattach.h
+===================================================================
+--- a/tools/hciattach.h	2015-09-03 20:19:36.000000000 -0500
++++ b/tools/hciattach.h	2020-03-04 20:31:39.000000000 -0600
+@@ -65,4 +65,4 @@
+ int qualcomm_init(int fd, int speed, struct termios *ti, const char *bdaddr);
+ int intel_init(int fd, int init_speed, int *speed, struct termios *ti);
+ int bcm43xx_init(int fd, int def_speed, int speed, struct termios *ti,
+-		const char *bdaddr);
++		const char *bdaddr, const char *fwdir);
+Index: b/tools/hciattach_bcm43xx.c
+===================================================================
+--- a/tools/hciattach_bcm43xx.c	2015-03-11 05:01:57.000000000 -0500
++++ b/tools/hciattach_bcm43xx.c	2020-03-08 20:39:18.603088384 -0500
+@@ -35,6 +35,7 @@
+ #include <dirent.h>
+ #include <time.h>
+ #include <limits.h>
++#include <strings.h>
+ 
+ #include "lib/bluetooth.h"
+ #include "lib/hci.h"
+@@ -42,10 +43,6 @@
+ 
+ #include "hciattach.h"
+ 
+-#ifndef FIRMWARE_DIR
+-#define FIRMWARE_DIR "/etc/firmware"
+-#endif
+-
+ #define FW_EXT ".hcd"
+ 
+ #define BCM43XX_CLOCK_48 1
+@@ -317,6 +314,7 @@
+ 		struct dirent *entry = readdir(dir);
+ 		if (!entry)
+ 			break;
++		fprintf(stderr, "Current %s\n", entry->d_name);
+ 
+ 		if (entry->d_type & DT_DIR) {
+ 			char path[PATH_MAX];
+@@ -329,14 +327,14 @@
+ 			ret = bcm43xx_locate_patch(path, chip_name, location);
+ 			if (!ret)
+ 				break;
+-		} else if (!strncmp(chip_name, entry->d_name, strlen(chip_name))) {
++		} else if (!strncasecmp(chip_name, entry->d_name, strlen(chip_name))) {
+ 			unsigned int name_len = strlen(entry->d_name);
+ 			size_t curs_ext = name_len - sizeof(FW_EXT) + 1;
+ 
+ 			if (curs_ext > name_len)
+ 				break;
+ 
+-			if (strncmp(FW_EXT, &entry->d_name[curs_ext], sizeof(FW_EXT)))
++			if (strncasecmp(FW_EXT, &entry->d_name[curs_ext], sizeof(FW_EXT)))
+ 				break;
+ 
+ 			/* found */
+@@ -352,20 +350,26 @@
+ }
+ 
+ int bcm43xx_init(int fd, int def_speed, int speed, struct termios *ti,
+-		const char *bdaddr)
++		const char *bdaddr, const char *fwdir)
+ {
+ 	char chip_name[20];
+ 	char fw_path[PATH_MAX];
+ 
+ 	printf("bcm43xx_init\n");
+ 
++	if (!fwdir) {
++		fwdir = "/etc/firmware";
++	}
```

Comment:
[Do not unnecessarily use braces where a single statement will do](https://www.kernel.org/doc/html/v4.10/process/coding-style.html#placing-braces-and-spaces)